### ISSUE TITLE
v12 RC1 external login provider not working

### ISSUE THREAD

**Issue by jbreuer** [umbraco/Umbraco-CMS#14265]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

12.0.0-rc1

### Bug summary

I upgraded the Umbraco OpenID Connect example package to `12.0.0-rc1` in this commit: https://github.com/jbreuer/Umbraco-OpenIdConnect-Example/commit/463771034db4d90bfbbe6591b25bee01dd6329ac

After the upgrade I try to login on the external login provider and I get the following error: 

`MissingMethodException: Method not found: 'Boolean Microsoft.IdentityModel.Tokens.TokenUtilities.IsRecoverableConfiguration(Microsoft.IdentityModel.Tokens.TokenValidationParameters, Microsoft.IdentityModel.Tokens.BaseConfiguration, Microsoft.IdentityModel.Tokens.BaseConfiguration ByRef)'.`

### Specifics

This seems to be the same error as reported here: https://github.com/umbraco/Umbraco-CMS/issues/14264. But this error happens on the frontend.

### Steps to reproduce

Go to this repo and branch: https://github.com/jbreuer/Umbraco-OpenIdConnect-Example/tree/feature/umbraco-12

Follow the instructions on this getting started video: https://www.youtube.com/watch?v=cklH7DtRDIQ

Once you have the project running go to the login page and press "Sign in with OpenIdConnect". Enter the credentials from the GitHub repo and you'll get this error:

```
An unhandled exception occurred while processing the request.
MissingMethodException: Method not found: 'Boolean Microsoft.IdentityModel.Tokens.TokenUtilities.IsRecoverableConfiguration(Microsoft.IdentityModel.Tokens.TokenValidationParameters, Microsoft.IdentityModel.Tokens.BaseConfiguration, Microsoft.IdentityModel.Tokens.BaseConfiguration ByRef)'.
System.IdentityModel.Tokens.Jwt.JwtSecurityTokenHandler.ValidateToken(string token, JwtSecurityToken outerToken, TokenValidationParameters validationParameters, out SecurityToken signatureValidatedToken)

Exception: An error was encountered while handling the remote login.
Microsoft.AspNetCore.Authentication.RemoteAuthenticationHandler<TOptions>.HandleRequestAsync()

Stack Query Cookies Headers Routing
MissingMethodException: Method not found: 'Boolean Microsoft.IdentityModel.Tokens.TokenUtilities.IsRecoverableConfiguration(Microsoft.IdentityModel.Tokens.TokenValidationParameters, Microsoft.IdentityModel.Tokens.BaseConfiguration, Microsoft.IdentityModel.Tokens.BaseConfiguration ByRef)'.
System.IdentityModel.Tokens.Jwt.JwtSecurityTokenHandler.ValidateToken(string token, JwtSecurityToken outerToken, TokenValidationParameters validationParameters, out SecurityToken signatureValidatedToken)
System.IdentityModel.Tokens.Jwt.JwtSecurityTokenHandler.ValidateToken(string token, TokenValidationParameters validationParameters, out SecurityToken validatedToken)
Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler.ValidateToken(string idToken, AuthenticationProperties properties, TokenValidationParameters validationParameters, out JwtSecurityToken jwt)
Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler.HandleRemoteAuthenticateAsync()

Show raw exception details
Exception: An error was encountered while handling the remote login.
Microsoft.AspNetCore.Authentication.RemoteAuthenticationHandler<TOptions>.HandleRequestAsync()
Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
StackExchange.Profiling.MiniProfilerMiddleware.Invoke(HttpContext context) in MiniProfilerMiddleware.cs
Umbraco.Cms.Web.Common.Middleware.UmbracoRequestMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
Umbraco.Cms.Web.Common.Middleware.UmbracoRequestMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
Microsoft.AspNetCore.Builder.UseMiddlewareExtensions+<>c__DisplayClass6_1+<<UseMiddlewareInterface>b__1>d.MoveNext()
Umbraco.Cms.Web.Common.Middleware.PreviewAuthenticationMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
Microsoft.AspNetCore.Builder.UseMiddlewareExtensions+<>c__DisplayClass6_1+<<UseMiddlewareInterface>b__1>d.MoveNext()
Umbraco.Cms.Web.Common.Middleware.UmbracoRequestLoggingMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
Microsoft.AspNetCore.Builder.UseMiddlewareExtensions+<>c__DisplayClass6_1+<<UseMiddlewareInterface>b__1>d.MoveNext()
SixLabors.ImageSharp.Web.Middleware.ImageSharpMiddleware.Invoke(HttpContext httpContext, bool retry)
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)

Show raw exception details
```

![image](https://github.com/umbraco/Umbraco-CMS/assets/5444597/087f489b-1bd3-446a-a1ad-f73e87ac1c56)

### Expected result / actual result

After upgrading to Umbraco 12 you should be able to login with an external login provider. I did the same upgrade from Umbraco 10 to Umbraco 11 and that just worked.

**Comment by bergmania** [comment]
My initial research points towards an issue introduced by in [OpenIddict 4.3](https://github.com/openiddict/openiddict-core/issues/1766) (They will[ downgrade again in 4.4](https://github.com/openiddict/openiddict-core/pull/1771))

The original error seems to be this https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2059

**Comment by nikolajlauridsen** [comment]
This has been fixed in #14279 and released in RC2 which is available on NuGet 😄

**Comment by jbreuer** [comment]
Thanks for fixing this. I upgraded to v12 RC2 and now login on the external login provider works again.
